beacon-chain/core/blocks/deposit.go

```diff
@@ -74,33 +83,41 @@ func ProcessDeposits(
 		return nil, err
 	}
 
+	// Attempt to verify all deposit signatures at once, if this fails then fall back to processing
+	// individual deposits with signature verification enabled.
 	deposits := b.Block.Body.Deposits
-	var err error
-	domain, err := helpers.ComputeDomain(params.BeaconConfig().DomainDeposit, nil, nil)
+	batchVerified, err := batchVerifyDepositsSignatures(ctx, deposits)
 	if err != nil {
 		return nil, err
 	}
 
-	// Attempt to verify all deposit signatures at once, if this fails then fall back to processing
-	// individual deposits with signature verification enabled.
-	var verifySignature bool
-	if err := verifyDepositDataWithDomain(ctx, deposits, domain); err != nil {
-		log.WithError(err).Debug("Failed to verify deposit data, verifying signatures individually")
-		verifySignature = true
-	}
-
 	for _, deposit := range deposits {
 		if deposit == nil || deposit.Data == nil {
 			return nil, errors.New("got a nil deposit in block")
 		}
-		beaconState, err = ProcessDeposit(beaconState, deposit, verifySignature)
+		beaconState, err = ProcessDeposit(beaconState, deposit, batchVerified)
 		if err != nil {
 			return nil, errors.Wrapf(err, "could not process deposit from %#x", bytesutil.Trunc(deposit.Data.PublicKey))
 		}
 	}
 	return beaconState, nil
 }
 
+func batchVerifyDepositsSignatures(ctx context.Context, deposits []*ethpb.Deposit) (bool, error) {
+	var err error
+	domain, err := helpers.ComputeDomain(params.BeaconConfig().DomainDeposit, nil, nil)
+	if err != nil {
+		return false, err
+	}
+
+	var verified bool
```

Comment:
This var name is pretty confusing  for its actual use case, as on a first read it would look like the signature is verified when in truth it means the opposite

Reply:
It's more standard than `verifySignature` which signals that the variable is actually a function.

Verified doesn't always imply true, it reads more as the act of verifying. ie. the state of being verified 

In our code base, we do use `verified` as boolean through out

One way to improve is to make it `verified := false`